app/assets/javascripts/discourse/app/components/search-menu.hbs

```diff
@@ -4,62 +4,78 @@
-        @action={{this.clearTopicContext}}
-        class="btn-small search-context"
-      />
-    {{else if this.inPMInboxContext}}
-      <DButton
-        @icon="xmark"
-        @label="search.in_messages"
-        @title="search.in_messages_tooltip"
-        @action={{this.clearPMInboxContext}}
-        class="btn-small search-context"
-      />
-    {{/if}}
+  <div class="search-input-wrapper">
+    <div class="search-input">
+      {{#if this.site.isMobileViewAndDevice}}
+        <SearchMenu::MobileSearchButton @onTap={{this.mobileSearch}} />
+      {{/if}}
 
-    <PluginOutlet
-      @name="search-menu-before-term-input"
-      @outletArgs={{hash openSearchMenu=this.open}}
-    />
+      {{#if this.search.inTopicContext}}
+        <DButton
+          @icon="xmark"
+          @label="search.in_this_topic"
+          @title="search.in_this_topic_tooltip"
+          @action={{this.clearTopicContext}}
+          class="btn-small search-context"
+        />
+      {{else if this.inPMInboxContext}}
+        <DButton
+          @icon="xmark"
+          @label="search.in_messages"
+          @title="search.in_messages_tooltip"
+          @action={{this.clearPMInboxContext}}
+          class="btn-small search-context"
+        />
+      {{/if}}
 
-    <SearchMenu::SearchTerm
-      @searchTermChanged={{this.searchTermChanged}}
-      @typeFilter={{this.typeFilter}}
-      @updateTypeFilter={{this.updateTypeFilter}}
-      @triggerSearch={{this.triggerSearch}}
-      @fullSearch={{this.fullSearch}}
-      @clearPMInboxContext={{this.clearPMInboxContext}}
-      @clearTopicContext={{this.clearTopicContext}}
-      @closeSearchMenu={{this.close}}
-      @openSearchMenu={{this.open}}
-      @autofocus={{@autofocusInput}}
-    />
+      <PluginOutlet
+        @name="search-menu-before-term-input"
+        @outletArgs={{hash openSearchMenu=this.open}}
+      />
 
-    {{#if this.loading}}
-      <div class="searching">
-        {{loading-spinner}}
-      </div>
-    {{else}}
-      <div class="searching">
-        <PluginOutlet @name="search-menu-before-advanced-search" />
-        {{#if this.search.activeGlobalSearchTerm}}
-          <SearchMenu::ClearButton @clearSearch={{this.clearSearch}} />
-        {{/if}}
-        <SearchMenu::AdvancedButton
-          @openAdvancedSearch={{this.openAdvancedSearch}}
-        />
-      </div>
+      <SearchMenu::SearchTerm
+        @searchTermChanged={{this.searchTermChanged}}
+        @typeFilter={{this.typeFilter}}
+        @updateTypeFilter={{this.updateTypeFilter}}
+        @triggerSearch={{this.triggerSearch}}
+        @fullSearch={{this.fullSearch}}
+        @clearPMInboxContext={{this.clearPMInboxContext}}
+        @clearTopicContext={{this.clearTopicContext}}
+        @closeSearchMenu={{this.close}}
+        @openSearchMenu={{this.open}}
+        @autofocus={{@autofocusInput}}
+        data-test-input="search-term"
+      />
+
+      {{#if this.loading}}
+        <div class="searching">
```

Comment:
we could share the div here

Reply:
@jjaffeux, you mean, share the .searching div between mobile and desktop view?